src/SFML/Graphics/Text.cpp

```diff
@@ -346,10 +346,13 @@ void Text::draw(RenderTarget& target, RenderStates states) const
 ////////////////////////////////////////////////////////////
 void Text::ensureGeometryUpdate() const
 {
-    // Do nothing, if geometry has not changed
-    if (!m_geometryNeedUpdate)
+    // Do nothing, if geometry has not changed and the font texture has not changed
+    if (!m_geometryNeedUpdate && m_font->getTexture(m_characterSize).m_cacheId == m_fontTextureId)
```

Comment:
In certain situations, this will dereference a NULL pointer causing undefined behaviour. The check for m_font is already done further down. Either move it up or move this down.